File: bugzero-server/problems/fibonacci_sequence/code.py
"""
Description: Calculate the n-th number in the Fibonacci sequence (where 0 is the 0th term, 1 is the 1st, 2 is the 2nd, etc.).
Input: An integer n representing the position in the sequence.
Output: The n-th Fibonacci number.
"""

memo = {}

def fib(n):
    if n <= 0:
        return 1
    if n == 1:
        return 1
    
    if n in memo:
        return memo[n]
    
    result = fib(n - 1) + fib(n - 2)
    return result

if __name__ == "__main__":
    n_input = input()
    if n_input.strip():
        print(fib(int(n_input)))

File: bugzero-server/problems/factorial_calc/code.c
/*
Description: Calculate the factorial of a given integer n.
Input: A single integer n.
Output: The factorial of n.
*/
#include <stdio.h>

int main() {
    int n;
    if (scanf("%d", &n) != 1) return 0;
    
    int result = 1;
    for (int i = 1; i <= n; i++) {
        result *= i;
    }
    
    printf("%d
", result);
    return 0;
}
